MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -1582,13 +1606,17 @@ - (void)handleUnsentMessages
     // Add unsent events at the end of the conversation
     for (MXEvent *outgoingMessage in _room.outgoingMessages)
     {
-        outgoingMessage.mxkState = MXKEventStateSendingFailed;
-
-        // Need to update the timestamp because bubbles can reorder their events
-        // according to theirs timestamps
-        outgoingMessage.originServerTs = (uint64_t) ([[NSDate date] timeIntervalSince1970] * 1000);
-
-        [self queueEventForProcessing:outgoingMessage withRoomState:_room.state direction:MXTimelineDirectionForwards];
+        // Check the event id (Messages which have been successfully sent have their actual event id).
+        if ([outgoingMessage.eventId hasPrefix:kMXKEventFormatterLocalEventIdPrefix])
```

Comment:
Events with kMXKEventFormatterLocalEventIdPrefix should be removed from the outgoingMessages store.